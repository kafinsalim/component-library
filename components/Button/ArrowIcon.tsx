import * as React from "react"

export function ArrowRight(props: any) {
  return (
    <svg id="arrow_right" width={19} height={14} viewBox="0 0 19 14" fill="none" {...props}>
      <path
        d="M.4 6.991l16.298-.03m0 0L11.252 1.61m5.447 5.351l-5.416 5.907"
        stroke="#fff"
        strokeWidth={2}
      />
    </svg>
  )
}

