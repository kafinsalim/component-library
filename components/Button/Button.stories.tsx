import * as React from 'react';
import { ThemeProvider } from 'emotion-theming';
import { boolean, text } from '@storybook/addon-knobs';
import decorators from '../../.storybook/utils';
import { theme as defaultTheme } from '../../utils';
import { Button } from '../';
import { SpacingBottom, DarkBackground } from '../shared/styles';

export default {
  title: 'Components/Button',
  decorators,
  parameters: {
    info: { 
      source: false,
      propTablesExclude: [SpacingBottom, DarkBackground, ThemeProvider]
     },
  },
};

export const Appereances = () => {
  return (
    <DarkBackground>
      <SpacingBottom>
        <Button> DEFAULT </Button>
      </SpacingBottom>
      <SpacingBottom>
        <Button color="primary"> PRIMARY </Button>
      </SpacingBottom>
      <SpacingBottom>
        <Button color="primary" variant="outline"> PRIMARY OUTLINE </Button>
      </SpacingBottom>
      <SpacingBottom>
        <Button color="secondary"> SECONDARY </Button>
      </SpacingBottom>
      <SpacingBottom>
        <Button color="secondary" variant="outline"> SECONDARY OUTLINE </Button>
      </SpacingBottom>
      <SpacingBottom>
        <Button color="primary" variant="icon"> â˜• </Button>
      </SpacingBottom>
      <SpacingBottom>
        <Button color="primary" variant="icon-circle"> â˜• </Button>
      </SpacingBottom>
      <SpacingBottom>
        <Button color="primary" variant="link"> OTHER </Button>
      </SpacingBottom>
    </DarkBackground>
  );
};

export const DisableState = () => {
  return (
    <DarkBackground>
      <SpacingBottom>
        <Button disabled> DEFAULT </Button>
      </SpacingBottom>
      <SpacingBottom>
        <Button disabled color="primary"> PRIMARY </Button>
      </SpacingBottom>
      <SpacingBottom>
        <Button disabled color="primary" variant="outline"> PRIMARY OUTLINE </Button>
      </SpacingBottom>
      <SpacingBottom>
        <Button disabled color="secondary"> SECONDARY </Button>
      </SpacingBottom>
      <SpacingBottom>
        <Button disabled color="secondary" variant="outline"> SECONDARY OUTLINE </Button>
      </SpacingBottom>
      <SpacingBottom>
        <Button disabled color="primary" variant="icon"> â˜• </Button>
      </SpacingBottom>
      <SpacingBottom>
        <Button disabled color="primary" variant="icon-circle"> â˜• </Button>
      </SpacingBottom>
    </DarkBackground>
  );
};

export const LoadingState = () => {
  return (
    <DarkBackground>
      <SpacingBottom>
        <Button loading> DEFAULT </Button>
      </SpacingBottom>
      <SpacingBottom>
        <Button loading color="primary"> PRIMARY </Button>
      </SpacingBottom>
      <SpacingBottom>
        <Button loading color="primary" variant="outline"> PRIMARY OUTLINE </Button>
      </SpacingBottom>
      <SpacingBottom>
        <Button loading color="secondary"> SECONDARY </Button>
      </SpacingBottom>
      <SpacingBottom>
        <Button loading color="secondary" variant="outline"> SECONDARY OUTLINE </Button>
      </SpacingBottom>
      <SpacingBottom>
        <Button loading color="primary" variant="icon"> â˜• </Button>
      </SpacingBottom>
      <SpacingBottom>
        <Button loading color="primary" variant="icon-circle"> â˜• </Button>
      </SpacingBottom>
    </DarkBackground>
  );
};

export const WithIcon = () => {
  return (
    <DarkBackground>
      <SpacingBottom>
        <Button startIcon={"ðŸ˜œ"} endIcon={"â–º"}> DEFAULT </Button>
      </SpacingBottom>
      <SpacingBottom>
        <Button startIcon={"ðŸ˜œ"} endIcon={"â–º"} color="primary"> PRIMARY </Button>
      </SpacingBottom>
      <SpacingBottom>
        <Button startIcon={"ðŸ˜œ"} endIcon={"â–º"} color="primary" variant="outline"> PRIMARY OUTLINE </Button>
      </SpacingBottom>
      <SpacingBottom>
        <Button startIcon={"ðŸ˜œ"} endIcon={"â–º"} color="secondary"> SECONDARY </Button>
      </SpacingBottom>
      <SpacingBottom>
        <Button startIcon={"ðŸ˜œ"} endIcon={"â–º"} color="secondary" variant="outline"> SECONDARY OUTLINE </Button>
      </SpacingBottom>
  </DarkBackground>
  )
};

export const Stressed = () => {
  return (
    <DarkBackground>
      <SpacingBottom>
        <Button> QWERTYASDFGHJKLSAJDJSADJHDJ SADGHSADHSADGJHASGDJHASGDJHSADHJSADASD ASDSAD </Button>
      </SpacingBottom>
      <SpacingBottom>
        <Button color="primary"> QWERTYASDFGHJKLSAJDJSADJHDJ SADGHSADHSADGJHASGDJHASGDJHSADHJSADASD ASDSAD </Button>
      </SpacingBottom>
      <SpacingBottom>
        <Button color="primary" variant="outline"> QWERTYASDFGHJKLSAJDJSADJHDJ SADGHSADHSADGJHASGDJHASGDJHSADHJSADASD ASDSAD </Button>
      </SpacingBottom>
      <SpacingBottom>
        <Button color="secondary"> QWERTYASDFGHJKLSAJDJSADJHDJ SADGHSADHSADGJHASGDJHASGDJHSADHJSADASD ASDSAD </Button>
      </SpacingBottom>
      <SpacingBottom>
        <Button color="secondary" variant="outline"> QWERTYASDFGHJKLSAJDJSADJHDJ SADGHSADHSADGJHASGDJHASGDJHSADHJSADASD ASDSAD </Button>
      </SpacingBottom>
      <SpacingBottom>
        <Button color="primary" variant="icon"> QWERTYASDFGHJKLSAJDJSADJHDJ SADGHSADHSADGJHASGDJHASGDJHSADHJSADASD ASDSAD </Button>
      </SpacingBottom>
      <SpacingBottom>
        <Button color="primary" variant="icon-circle"> QWERTYASDFGHJKLSAJDJSADJHDJ SADGHSADHSADGJHASGDJHASGDJHSADHJSADASD ASDSAD </Button>
      </SpacingBottom>
  </DarkBackground>
  )
};

const theme = {
  ...defaultTheme,
  palette: {
    ...defaultTheme.palette,
    orange: {
      main: '#F34706',
      light: '#FF5964',
      contrastText: '#fff'
    },
  },
};

export const OverwriteWithTheme = () => {
  return (
    <ThemeProvider theme={theme}>
      <DarkBackground>
        <Button
          variant={text('variant', 'solid')}
          color={text('color', 'orange')}
          size={text('size', 'medium')}
          disabled={boolean('disabled', false)}
          enableElevation={boolean('enableElevation', false)}
        >
          ORANGE
        </Button>
      </DarkBackground>
    </ThemeProvider>
  );
};

