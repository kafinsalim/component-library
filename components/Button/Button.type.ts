import {
  ITheme,
} from '../../utils';

export type buttonPropSize = {
  [key: string]: any
}

export type variantType = {
  [key: string]: any
}

export type PropsByVariant = {
  variant: string,
  color: string,
  theme: ITheme,
}

export interface IButtonProps {
  color?: string,
  size?: string,
  variant?: string,
  enableElevation?: boolean,
  disabled?: boolean,
  theme?: any,
  onClick?: any,
  children: React.ReactNode,
  startIcon?: React.ReactNode,
  endIcon?: React.ReactNode,
  loading?: boolean
}

export type iconWrapperProp = {
  leftSpacing?: boolean,
  rightSpacing?: boolean,
  isSecondary?: boolean
}