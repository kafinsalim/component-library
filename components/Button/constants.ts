export const COLOR_SECONDARY = 'secondary'