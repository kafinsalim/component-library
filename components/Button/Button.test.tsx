import * as React from 'react';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event'
import { Button } from './Button';
import serializer, { matchers } from 'jest-emotion'
import { spacing, theme } from '../../utils';

expect.addSnapshotSerializer(serializer)
expect.extend(matchers)

describe('<Button />', () => {
  it('should render the default button with a text', () => {
    render(<Button>Default Button</Button>)

    expect(screen.getByText('Default Button')).toBeInTheDocument()
    expect(screen.getByRole('button')).toMatchSnapshot()
  });

  it('should render the primary button', () => {
    render(<Button color="primary">Primary</Button>)

    expect(screen.getByText('Primary')).toBeInTheDocument()
    expect(screen.getByRole('button')).toHaveStyleRule('min-height', '48px')
    expect(screen.getByRole('button')).toHaveStyleRule('border-radius', '10px')
    expect(screen.getByRole('button')).toHaveStyleRule('padding', `${spacing["small"]} ${spacing["gutter"]}`)
    expect(screen.getByRole('button')).toHaveStyleRule('background-color', theme.palette.primary.main)
    expect(screen.getByRole('button')).toMatchSnapshot()
  });

  it('should render the primary button with outline', () => {
    render(<Button color="primary" variant="outline">Primary</Button>)

    expect(screen.getByText('Primary')).toBeInTheDocument()
    expect(screen.getByRole('button')).toHaveStyleRule('min-height', '48px')
    expect(screen.getByRole('button')).toHaveStyleRule('border-radius', '10px')
    expect(screen.getByRole('button')).toHaveStyleRule('padding', `${spacing["small"]} ${spacing["gutter"]}`)
    expect(screen.getByRole('button')).toHaveStyleRule('background-color', 'transparent')
    expect(screen.getByRole('button')).toHaveStyleRule('border', `3px solid ${theme.palette.yellow.main}`)
    expect(screen.getByRole('button')).toMatchSnapshot()
  });

  it('should render the secondary button', () => {
    render(<Button color="secondary">Secondary</Button>)

    expect(screen.getByText('Secondary')).toBeInTheDocument()
    expect(screen.getByRole('button')).toHaveStyleRule('min-height', '42px')
    expect(screen.getByRole('button')).toHaveStyleRule('border-radius', '10px')
    expect(screen.getByRole('button')).toHaveStyleRule('padding', `${spacing["small"]} ${spacing["gutter"]}`)
    expect(screen.getByRole('button')).toHaveStyleRule('background-color', '#494B50')
    expect(screen.getByRole('button')).toMatchSnapshot()
  });

  it('should render the secondary button with outline', () => {
    render(<Button color="secondary" variant="outline">Secondary</Button>)

    expect(screen.getByText('Secondary')).toBeInTheDocument()
    expect(screen.getByRole('button')).toHaveStyleRule('min-height', '42px')
    expect(screen.getByRole('button')).toHaveStyleRule('border-radius', '10px')
    expect(screen.getByRole('button')).toHaveStyleRule('padding', `${spacing["small"]} ${spacing["gutter"]}`)
    expect(screen.getByRole('button')).toHaveStyleRule('background-color', 'rgba(185,184,185,0.1)')
    expect(screen.getByRole('button')).toHaveStyleRule('border', `2px solid ${theme.palette.common.grey}`)
    expect(screen.getByRole('button')).toMatchSnapshot()
  });
  
  it('should render the variant="icon"', () => {
    render(<Button color="primary" variant="icon">icon</Button>)
    
    expect(screen.getByText('icon')).toBeInTheDocument()
    expect(screen.getByRole('button')).toHaveStyleRule('background-color', theme.palette.primary.main)
    expect(screen.getByRole('button')).toHaveStyleRule('border-radius', '10px')
    expect(screen.getByRole('button')).toHaveStyleRule('padding', '0 !important')
    expect(screen.getByRole('button')).toHaveStyleRule('width', '48px')
    expect(screen.getByRole('button')).toHaveStyleRule('min-height', '48px')
    expect(screen.getByRole('button')).toMatchSnapshot()
  });

  it('should render the variant="icon-circle"', () => {
    render(<Button color="primary" variant="icon-circle">icon circle</Button>)

    expect(screen.getByText('icon circle')).toBeInTheDocument()
    expect(screen.getByRole('button')).toHaveStyleRule('border-radius', '50%')
    expect(screen.getByRole('button')).toHaveStyleRule('width', '48px')
    expect(screen.getByRole('button')).toHaveStyleRule('min-height', '48px')
    expect(screen.getByRole('button')).toMatchSnapshot()
  });

  it('should fired on click', () => {
    const onClick = jest.fn()
    render(<Button onClick={onClick}>Default Button</Button>)

    // click one time
    userEvent.click(screen.getByText('Default Button'))
    expect(onClick).toHaveBeenCalledTimes(1)

    // click two times
    userEvent.click(screen.getByText('Default Button'))
    expect(onClick).toHaveBeenCalledTimes(2)
  });


  it('should render with disabled attribute & cannot be click', () => {
    const onClick = jest.fn()
    render(<Button onClick={onClick} size="medium" disabled>With disabled</Button>)

    expect(screen.getByRole('button')).toBeDisabled()
    userEvent.click(screen.getByRole('button'))
    // cannot be clicked when disabled
    expect(onClick).not.toHaveBeenCalled()
    expect(screen.getByRole('button')).toMatchSnapshot()
  });

  it('should render with loading spinner', () => {
    const wrapper = render(<Button loading>With disabled</Button>)
    const svgSpinnerLoading = wrapper.container.getElementsByTagName('svg')
    
    expect(svgSpinnerLoading[0]).toBeInTheDocument()
    expect(screen.getByRole('button')).toMatchSnapshot()
  });

  it('should render with enable elevation', () => {
    render(<Button size="medium" enableElevation>With disabled</Button>)

    expect(screen.getByRole('button')).toHaveStyleRule('box-shadow', theme.shadows[1])
    expect(screen.getByRole('button')).toMatchSnapshot()
  });
});