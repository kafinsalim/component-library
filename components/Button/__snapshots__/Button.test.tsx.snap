// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render the default button with a text 1`] = `
.emotion-0 {
  min-height: 48px;
  min-width: 96px;
  font-weight: 500;
  cursor: pointer;
  -webkit-transition: all 0.15s linear;
  transition: all 0.15s linear;
  -webkit-letter-spacing: 0.6px;
  -moz-letter-spacing: 0.6px;
  -ms-letter-spacing: 0.6px;
  letter-spacing: 0.6px;
  padding: 0.75rem 1.5rem;
  border-radius: 10px;
  font-family: 'Rubik',sans-serif;
  outline: 0;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-shadow: 0px 4px 15px rgba(0,0,0,0.25);
  border: 1px solid #EAEAEA;
  background-color: #EAEAEA;
  color: #111;
  font-size: 0.75rem;
}

.emotion-0:hover {
  border: 1px solid #C9C5C5;
  background-color: #C9C5C5;
}

.emotion-0:disabled {
  cursor: not-allowed;
}

.emotion-0 > :first-of-type:last-of-type {
  margin: 0 auto;
}

@media (min-width:600px) {
  .emotion-0 {
    min-height: 52px;
    font-size: 0.825rem;
  }
}

<button
  class="emotion-0"
>
  Default Button
</button>
`;

exports[`<Button /> should render the primary button 1`] = `
.emotion-0 {
  min-height: 48px;
  min-width: 96px;
  font-weight: 500;
  cursor: pointer;
  -webkit-transition: all 0.15s linear;
  transition: all 0.15s linear;
  -webkit-letter-spacing: 0.6px;
  -moz-letter-spacing: 0.6px;
  -ms-letter-spacing: 0.6px;
  letter-spacing: 0.6px;
  padding: 0.75rem 1.5rem;
  border-radius: 10px;
  font-family: 'Rubik',sans-serif;
  outline: 0;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-shadow: 0px 4px 15px rgba(0,0,0,0.25);
  border: 2px solid transparent;
  background-color: #494B50;
  color: white;
  font-size: 0.75rem;
}

.emotion-0:hover {
  border: 2px solid transparent;
  background-color: #F2CC23;
  color: #494B50;
}

.emotion-0:active {
  border: 2px solid transparent;
  background-color: #E3BF22;
  color: #2D2F33;
}

.emotion-0:focus {
  border: 2px solid rgba(255,255,255,0.75);
  background-color: #494B50;
  color: white;
}

.emotion-0:disabled {
  cursor: not-allowed;
  border: 2px solid transparent;
  background-color: #888;
  color: rgba(255,255,255,0.5);
}

.emotion-0 > :first-of-type:last-of-type {
  margin: 0 auto;
}

@media (min-width:600px) {
  .emotion-0 {
    min-height: 52px;
    font-size: 0.825rem;
  }
}

<button
  class="emotion-0"
>
  Primary
</button>
`;

exports[`<Button /> should render the primary button with outline 1`] = `
.emotion-0 {
  min-height: 48px;
  min-width: 96px;
  font-weight: 500;
  cursor: pointer;
  -webkit-transition: all 0.15s linear;
  transition: all 0.15s linear;
  -webkit-letter-spacing: 0.6px;
  -moz-letter-spacing: 0.6px;
  -ms-letter-spacing: 0.6px;
  letter-spacing: 0.6px;
  padding: 0.75rem 1.5rem;
  border-radius: 10px;
  font-family: 'Rubik',sans-serif;
  outline: 0;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-shadow: 0px 4px 15px rgba(0,0,0,0.25);
  border: 3px solid #F2CC23;
  background-color: transparent;
  color: #F2CC23;
  font-size: 0.75rem;
}

.emotion-0 #loading_spinner_animated > path {
  fill: #494B50;
}

.emotion-0:hover {
  border: 3px solid #494B50;
  color: #494B50;
}

.emotion-0:active {
  border: 3px solid #E3BF22;
  color: #2D2F33;
}

.emotion-0:focus {
  border: 3px solid #888;
  color: white;
}

.emotion-0:disabled {
  cursor: not-allowed;
  border: 3px solid #888;
  color: #888;
}

.emotion-0 > :first-of-type:last-of-type {
  margin: 0 auto;
}

@media (min-width:600px) {
  .emotion-0 {
    min-height: 52px;
    font-size: 0.825rem;
  }
}

<button
  class="emotion-0"
>
  Primary
</button>
`;

exports[`<Button /> should render the secondary button 1`] = `
.emotion-1 {
  min-height: 42px;
  min-width: 96px;
  font-weight: 500;
  cursor: pointer;
  -webkit-transition: all 0.15s linear;
  transition: all 0.15s linear;
  -webkit-letter-spacing: 0.6px;
  -moz-letter-spacing: 0.6px;
  -ms-letter-spacing: 0.6px;
  letter-spacing: 0.6px;
  padding: 0.75rem 1.5rem;
  border-radius: 10px;
  font-family: 'Rubik',sans-serif;
  outline: 0;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-shadow: 0px 4px 15px rgba(0,0,0,0.25);
  border: 2px solid transparent;
  background-color: #494B50;
  color: white;
  font-size: 0.75rem;
}

.emotion-1 #arrow_right {
  position: relative;
  left: 0;
  -webkit-transition: all ease-in 0.2s;
  transition: all ease-in 0.2s;
}

.emotion-1:hover {
  border: 2px solid transparent;
  color: white;
  background-color: rgba(255,255,255,0.25);
}

.emotion-1:hover #arrow_right {
  left: 8px;
}

.emotion-1:disabled {
  cursor: not-allowed;
}

.emotion-1 > :first-of-type:last-of-type {
  margin: 0 auto;
}

@media (min-width:600px) {
  .emotion-1 {
    min-height: 52px;
    font-size: 0.825rem;
  }
}

.emotion-0 {
  padding-left: 0.5rem;
  padding-right: 0;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

@media (min-width:600px) {
  .emotion-0 {
    padding-left: 3rem;
  }
}

<button
  class="emotion-1"
>
  Secondary
  <span
    class="emotion-0"
  >
    <svg
      fill="none"
      height="14"
      id="arrow_right"
      viewBox="0 0 19 14"
      width="19"
    >
      <path
        d="M.4 6.991l16.298-.03m0 0L11.252 1.61m5.447 5.351l-5.416 5.907"
        stroke="#fff"
        stroke-width="2"
      />
    </svg>
  </span>
</button>
`;

exports[`<Button /> should render the secondary button with outline 1`] = `
.emotion-0 {
  padding-left: 0.5rem;
  padding-right: 0;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

@media (min-width:600px) {
  .emotion-0 {
    padding-left: 3rem;
  }
}

.emotion-1 {
  min-height: 42px;
  min-width: 96px;
  font-weight: 500;
  cursor: pointer;
  -webkit-transition: all 0.15s linear;
  transition: all 0.15s linear;
  -webkit-letter-spacing: 0.6px;
  -moz-letter-spacing: 0.6px;
  -ms-letter-spacing: 0.6px;
  letter-spacing: 0.6px;
  padding: 0.75rem 1.5rem;
  border-radius: 10px;
  font-family: 'Rubik',sans-serif;
  outline: 0;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-shadow: 0px 4px 15px rgba(0,0,0,0.25);
  border: 2px solid #B9B8B9;
  background-color: rgba(185,184,185,0.1);
  color: white;
  font-size: 0.75rem;
}

.emotion-1 #arrow_right {
  position: relative;
  left: 0;
  -webkit-transition: all ease-in 0.2s;
  transition: all ease-in 0.2s;
}

.emotion-1:hover {
  border: 2px solid white;
  color: white;
  background-color: #52565C;
}

.emotion-1:hover #arrow_right {
  left: 8px;
}

.emotion-1:disabled {
  cursor: not-allowed;
}

.emotion-1 > :first-of-type:last-of-type {
  margin: 0 auto;
}

@media (min-width:600px) {
  .emotion-1 {
    min-height: 52px;
    font-size: 0.825rem;
  }
}

<button
  class="emotion-1"
>
  Secondary
  <span
    class="emotion-0"
  >
    <svg
      fill="none"
      height="14"
      id="arrow_right"
      viewBox="0 0 19 14"
      width="19"
    >
      <path
        d="M.4 6.991l16.298-.03m0 0L11.252 1.61m5.447 5.351l-5.416 5.907"
        stroke="#fff"
        stroke-width="2"
      />
    </svg>
  </span>
</button>
`;

exports[`<Button /> should render the variant="icon" 1`] = `
.emotion-0 {
  min-height: 48px;
  min-width: 48px;
  font-weight: 500;
  cursor: pointer;
  -webkit-transition: all 0.15s linear;
  transition: all 0.15s linear;
  -webkit-letter-spacing: 0.6px;
  -moz-letter-spacing: 0.6px;
  -ms-letter-spacing: 0.6px;
  letter-spacing: 0.6px;
  padding: 0 !important;
  border-radius: 10px;
  font-family: 'Rubik',sans-serif;
  outline: 0;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-shadow: 0px 4px 15px rgba(0,0,0,0.25);
  border: 2px solid transparent;
  background-color: #494B50;
  color: white;
  width: 48px;
  font-size: 0.75rem;
}

.emotion-0:hover {
  border: 2px solid transparent;
  background-color: #F2CC23;
  color: #494B50;
}

.emotion-0:active {
  border: 2px solid transparent;
  background-color: #E3BF22;
  color: #2D2F33;
}

.emotion-0:focus {
  border: 2px solid rgba(255,255,255,0.75);
  background-color: #494B50;
  color: white;
}

.emotion-0:disabled {
  cursor: not-allowed;
  border: 2px solid transparent;
  background-color: #888;
  color: rgba(255,255,255,0.5);
}

.emotion-0 > :first-of-type:last-of-type {
  margin: 0 auto;
}

@media (min-width:600px) {
  .emotion-0 {
    min-height: 52px;
    font-size: 0.825rem;
    width: 52px;
    min-width: 52px;
  }
}

<button
  class="emotion-0"
>
  icon
</button>
`;

exports[`<Button /> should render the variant="icon-circle" 1`] = `
.emotion-0 {
  min-height: 48px;
  min-width: 48px;
  font-weight: 500;
  cursor: pointer;
  -webkit-transition: all 0.15s linear;
  transition: all 0.15s linear;
  -webkit-letter-spacing: 0.6px;
  -moz-letter-spacing: 0.6px;
  -ms-letter-spacing: 0.6px;
  letter-spacing: 0.6px;
  padding: 0 !important;
  border-radius: 50%;
  font-family: 'Rubik',sans-serif;
  outline: 0;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-shadow: 0px 4px 15px rgba(0,0,0,0.25);
  border: 2px solid transparent;
  background-color: #494B50;
  color: white;
  width: 48px;
  font-size: 0.75rem;
}

.emotion-0:hover {
  border: 2px solid transparent;
  background-color: #F2CC23;
  color: #494B50;
}

.emotion-0:active {
  border: 2px solid transparent;
  background-color: #E3BF22;
  color: #2D2F33;
}

.emotion-0:focus {
  border: 2px solid rgba(255,255,255,0.75);
  background-color: #494B50;
  color: white;
}

.emotion-0:disabled {
  cursor: not-allowed;
  border: 2px solid transparent;
  background-color: #888;
  color: rgba(255,255,255,0.5);
}

.emotion-0 > :first-of-type:last-of-type {
  margin: 0 auto;
}

@media (min-width:600px) {
  .emotion-0 {
    min-height: 52px;
    font-size: 0.825rem;
    width: 52px;
    min-width: 52px;
  }
}

<button
  class="emotion-0"
>
  icon circle
</button>
`;

exports[`<Button /> should render with disabled attribute & cannot be click 1`] = `
.emotion-0 {
  min-height: 48px;
  min-width: 96px;
  font-weight: 500;
  cursor: pointer;
  opacity: 0.7;
  -webkit-transition: all 0.15s linear;
  transition: all 0.15s linear;
  -webkit-letter-spacing: 0.6px;
  -moz-letter-spacing: 0.6px;
  -ms-letter-spacing: 0.6px;
  letter-spacing: 0.6px;
  padding: 0.75rem 1.5rem;
  border-radius: 10px;
  font-family: 'Rubik',sans-serif;
  outline: 0;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-shadow: 0px 4px 15px rgba(0,0,0,0.25);
  border: 1px solid #EAEAEA;
  background-color: #EAEAEA;
  color: #111;
  font-size: 0.75rem;
}

.emotion-0:disabled {
  cursor: not-allowed;
}

.emotion-0 > :first-of-type:last-of-type {
  margin: 0 auto;
}

@media (min-width:600px) {
  .emotion-0 {
    min-height: 52px;
    font-size: 0.825rem;
  }
}

<button
  class="emotion-0"
  disabled=""
>
  With disabled
</button>
`;

exports[`<Button /> should render with enable elevation 1`] = `
.emotion-0 {
  min-height: 48px;
  min-width: 96px;
  font-weight: 500;
  cursor: pointer;
  -webkit-transition: all 0.15s linear;
  transition: all 0.15s linear;
  -webkit-letter-spacing: 0.6px;
  -moz-letter-spacing: 0.6px;
  -ms-letter-spacing: 0.6px;
  letter-spacing: 0.6px;
  padding: 0.75rem 1.5rem;
  border-radius: 10px;
  font-family: 'Rubik',sans-serif;
  outline: 0;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-shadow: 0px 5px 10px rgba(0,0,0,0.12);
  border: 1px solid #EAEAEA;
  background-color: #EAEAEA;
  color: #111;
  font-size: 0.75rem;
}

.emotion-0:hover {
  box-shadow: 0px 8px 30px rgba(0,0,0,0.24);
  border: 1px solid #C9C5C5;
  background-color: #C9C5C5;
}

.emotion-0:active {
  box-shadow: 0px 8px 30px rgba(0,0,0,0.24);
}

.emotion-0:disabled {
  cursor: not-allowed;
}

.emotion-0 > :first-of-type:last-of-type {
  margin: 0 auto;
}

@media (min-width:600px) {
  .emotion-0 {
    min-height: 52px;
    font-size: 0.825rem;
  }
}

<button
  class="emotion-0"
>
  With disabled
</button>
`;

exports[`<Button /> should render with loading spinner 1`] = `
@keyframes animation-0 {
  0% {
    -webkit-transform: rotate(0deg);
    -ms-transform: rotate(0deg);
    transform: rotate(0deg);
  }

  100% {
    -webkit-transform: rotate(360deg);
    -ms-transform: rotate(360deg);
    transform: rotate(360deg);
  }
}

.emotion-1 {
  min-height: 48px;
  min-width: 96px;
  font-weight: 500;
  cursor: pointer;
  -webkit-transition: all 0.15s linear;
  transition: all 0.15s linear;
  -webkit-letter-spacing: 0.6px;
  -moz-letter-spacing: 0.6px;
  -ms-letter-spacing: 0.6px;
  letter-spacing: 0.6px;
  padding: 0.75rem 1.5rem;
  border-radius: 10px;
  font-family: 'Rubik',sans-serif;
  outline: 0;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-shadow: 0px 4px 15px rgba(0,0,0,0.25);
  border: 1px solid #EAEAEA;
  background-color: #EAEAEA;
  color: #111;
  font-size: 0.75rem;
}

.emotion-1:hover {
  border: 1px solid #C9C5C5;
  background-color: #C9C5C5;
}

.emotion-1:disabled {
  cursor: not-allowed;
}

.emotion-1 > :first-of-type:last-of-type {
  margin: 0 auto;
}

@media (min-width:600px) {
  .emotion-1 {
    min-height: 52px;
    font-size: 0.825rem;
  }
}

.emotion-0 {
  -webkit-animation: animation-0 1s linear infinite;
  animation: animation-0 1s linear infinite;
}

<button
  class="emotion-1"
>
  <svg
    class="emotion-0"
    fill="none"
    height="24"
    id="loading_spinner_animated"
    viewBox="0 0 24 24"
    width="24"
  >
    <path
      d="M21.477 11.424a.922.922 0 00-1.843.086 7.366 7.366 0 11-3.38-5.853.922.922 0 10.996-1.553 9.208 9.208 0 104.227 7.32z"
      fill="#fff"
    />
  </svg>
</button>
`;
